core/src/test/java/org/infinispan/xsite/irac/Irac3SitesConflictTest.java

```diff
@@ -30,32 +37,159 @@
    private static final int CLUSTER_SIZE = 3;
    private final List<ManualIracManager> iracManagerList;
 
+   private ConfigMode configMode;
+
+   public Irac3SitesConflictTest configMode(ConfigMode configMode) {
+      this.configMode = configMode;
+      return this;
+   }
+
+   private enum ConfigMode {
+      NON_TX,
+      PESSIMISTIC_TX,
+      OPTIMISTIC_TX_RC,
+      OPTIMISTIC_TX_RR,
+   }
+
+   @Factory
+   public Object[] factory() {
+      List<Irac3SitesConflictTest> tests = new ArrayList<>();
+      for (ConfigMode configMode : ConfigMode.values()) {
+         tests.add(new Irac3SitesConflictTest().configMode(configMode));
+      }
+      return tests.toArray();
+   }
+
+   @Override
+   protected String[] parameterNames() {
+      return new String[]{"configMode"};
+   }
+
+   @Override
+   protected Object[] parameterValues() {
+      return new Object[]{configMode};
+   }
+
    protected Irac3SitesConflictTest() {
       this.iracManagerList = new ArrayList<>(N_SITES * CLUSTER_SIZE);
    }
 
    public void testPutIfAbsent(Method method) {
-      doTest(method, TestOperation.PUT_IF_ABSENT);
+      doTest(method, new TestOperationInterop(TestOperation.PUT_IF_ABSENT));
    }
 
    public void testPut(Method method) {
-      doTest(method, TestOperation.PUT);
+      doTest(method, new TestOperationInterop(TestOperation.PUT));
    }
 
    public void testReplace(Method method) {
-      doTest(method, TestOperation.REPLACE);
+      doTest(method, new TestOperationInterop(TestOperation.REPLACE));
    }
 
    public void testConditionalReplace(Method method) {
-      doTest(method, TestOperation.REPLACE_CONDITIONAL);
+      doTest(method, new TestOperationInterop(TestOperation.REPLACE_CONDITIONAL));
    }
 
    public void testRemove(Method method) {
-      doTest(method, TestOperation.REMOVE);
+      doTest(method, new TestOperationInterop(TestOperation.REMOVE));
    }
 
    public void testConditionalRemove(Method method) {
-      doTest(method, TestOperation.REMOVE_CONDITIONAL);
+      doTest(method, new TestOperationInterop(TestOperation.REMOVE_CONDITIONAL));
+   }
+
+   // TODO: need to do this still?
+   //   public void testMaxIdleExpirationASync(Method method) {
```

Comment:
what does sync/async mean here? the xsite is async only in this class.

Reply:
I meant the remove expired command itself. The test waits for its completion. If it didn't then we could have a remove expired and other puts running concurrently.

Reply:
I'm not sure if it matters. the IracManager is blocking everything until the end. You don't have the remove/put going to the other sites.

Reply:
It shouldn't block the invoking thread. It returns a CompletionStage.